src/hipscat_import/index/map_reduce.py

```diff
@@ -31,16 +31,38 @@ def create_index(args):
     )
 
     if args.include_order_pixel:
-        ## Take out the hive dictionary behavior.
+        ## Take out the hive dictionary behavior that turns these into int32.
         data["Norder"] = data["Norder"].astype(np.uint8)
         data["Dir"] = data["Dir"].astype(np.uint64)
         data["Npix"] = data["Npix"].astype(np.uint64)
+
+    # There are some silly dask things happening here:
+    # - Turn the existing index column into a regular column
+    # - If that had been the _hipscat_index, and we don't want it anymore, drop it
+    # - Create a new index, using our target indexing_column.
+    #   Use division hints if provided.
+    # - Then re-partition the whole thing according to compute size,
+    #   because the following de-duplication step requires smaller chunks.
+    # - Drop duplicate rows (but reset the index so it can be included in
+    #   duplicate-finding)
     data = data.reset_index()
     if not args.include_hipscat_index:
         data = data.drop(columns=[HIPSCAT_ID_COLUMN])
-    data = data.drop_duplicates()
-    data = data.repartition(partition_size=args.compute_partition_size)
-    data = data.set_index(args.indexing_column)
+
+    if args.divisions is not None and len(args.divisions) > 2:
+        data = data.set_index(args.indexing_column, divisions=args.divisions)
+    else:
+        data = data.set_index(args.indexing_column)
```

Comment:
Just in case you aren't aware, with default parameters this set_index call will be sorting by the new index column as well.  https://docs.dask.org/en/latest/generated/dask.dataframe.DataFrame.set_index.html

Reply:
Yes! It just won't have the divisions to use as hints, and so will have much worse performance for large data sets.